admin/homebrew.py

```diff
@@ -108,68 +94,78 @@ def get_resource_stanzas(dependency_graph):
+def make_recipe(version, sdist_url):
+    """
+    Create a Homebrew recipe. This uses the network.
 
-    If the command is called with a single argument, the argument
-    provides a URL to retrieve the initial source distribution archive.
+    :param version: The version of Flocker to create a recipe for.
+    :param sdist_url: The URL of the source distribution of Flocker.
 
-    If no command line argument is provided, use the standard release
-    location for the indicated version.
+    :return unicode: A Homebrew recipe.
     """
-    logging.basicConfig(level=logging.INFO)
+    dependency_graph = get_dependency_graph(u"flocker")
+    return get_recipe(
+        sdist_url=sdist_url,
+        sha1=get_checksum(url=sdist_url),
+        class_name=get_class_name(version=version),
+        resources=get_resources(dependency_graph=dependency_graph),
+    )
 
-    parser = argparse.ArgumentParser(
-        description='Create a Homebrew recipe from a source distribution.')
-    parser.add_argument(
-        '--flocker-version', help='version number for the Homebrew recipe'
-        ' (either this argument or the environment variable VERSION must be'
-        ' provided)')
-    parser.add_argument(
-        '--sdist', help='URL of the source distribution')
-    parser.add_argument(
-        '--output-file',
-        help='filename for created Homebrew recipe (default: stdout)')
-    args = parser.parse_args()
-
-    # If version not supplied, for backwards-compatibility, get it from
-    # the environment variable VERSION.
-    version = args.flocker_version
-    if version is None:
-        version = environ.get('VERSION')
-        if version is None:
-            parser.print_help()
-            sys.exit(1)
-    logging.info('Creating Homebrew recipe for version {}'.format(version))
 
-    # If url not supplied, for backwards-compatibility, use the Google
-    # Storage location specified in the release process.
-    url = args.sdist
-    if url is None:
-        url = (b"https://clusterhq-archive.s3.amazonaws.com/python/"
-               "Flocker-{version}.tar.gz").format(version=version)
+def get_recipe(sdist_url, sha1, class_name, resources):
+    """
+    Create a Homebrew recipe. This does not use the network.
 
-    dependency_graph = get_dependency_graph(u"flocker")
+    :param sdist_url: The URL of the source distribution of Flocker.
+    :param sha1: The checksum of ``sdist_url``.
+    :param class_name: The recipe Ruby class name.
+    :param resources: List of dictionaries mapping project names to urls and
+        checksums or dependencies.
+
+    :return unicode: A Homebrew recipe.
+    """
+    dependencies = [resource['project_name'] for resource in resources]
 
-    recipe = u"""require "formula"
+    return u"""require "formula"
 
 class {class_name} < Formula
   homepage "https://clusterhq.com"
-  url "{url}"
+  url "{sdist_url}"
   sha1 "{sha1}"
   depends_on :python if MacOS.version <= :snow_leopard
 {resources}
```

Comment:
I'm not sure what this refers to. If I'm guessing I'd say it is because GitHub displays a space before `{resources}` which is not present in the source.

Anyways, the tests cover the indentation.